modules/dnn/src/dnn.cpp

```diff
@@ -97,35 +97,38 @@ namespace
 }
 
 Mat blobFromImage(InputArray image, double scalefactor, const Size& size,
-                  const Scalar& mean, bool swapRB, bool crop)
+                  const Scalar& mean, bool swapRB, bool crop, int type)
 {
     CV_TRACE_FUNCTION();
     Mat blob;
-    blobFromImage(image, blob, scalefactor, size, mean, swapRB, crop);
+    blobFromImage(image, blob, scalefactor, size, mean, swapRB, crop, type);
     return blob;
 }
 
 void blobFromImage(InputArray image, OutputArray blob, double scalefactor,
-                   const Size& size, const Scalar& mean, bool swapRB, bool crop)
+                   const Size& size, const Scalar& mean, bool swapRB, bool crop, int type)
 {
     CV_TRACE_FUNCTION();
     std::vector<Mat> images(1, image.getMat());
-    blobFromImages(images, blob, scalefactor, size, mean, swapRB, crop);
+    blobFromImages(images, blob, scalefactor, size, mean, swapRB, crop, type);
 }
 
 Mat blobFromImages(InputArrayOfArrays images, double scalefactor, Size size,
-                   const Scalar& mean, bool swapRB, bool crop)
+                   const Scalar& mean, bool swapRB, bool crop, int type)
 {
     CV_TRACE_FUNCTION();
     Mat blob;
-    blobFromImages(images, blob, scalefactor, size, mean, swapRB, crop);
+    blobFromImages(images, blob, scalefactor, size, mean, swapRB, crop, type);
     return blob;
 }
 
 void blobFromImages(InputArrayOfArrays images_, OutputArray blob_, double scalefactor,
-                    Size size, const Scalar& mean_, bool swapRB, bool crop)
+                    Size size, const Scalar& mean_, bool swapRB, bool crop, int type)
 {
     CV_TRACE_FUNCTION();
+    CV_Assert(type == CV_32F ||
+              type == CV_8U && mean_ == Scalar() && scalefactor == 1.0);
```

Comment:
Looks like there is () hell.

Please use `CV_CheckType` macro (we should avoid messages like [this](https://github.com/opencv/opencv/issues/12012#issue-342551071)):
```
CV_CheckType(type, type == CV_32F || type == CV_8U, "Input type should be CV_32F or CV_8U");
if (type == CV_8U)
{
    CV_CheckEQ(scalefactor, 1.0, "Scale is not supported for 8U inputs");
    CV_Assert(mean_ == Scalar() && "Mean shift is not supported for 8U inputs"); // TODO Need check for this
}
```